bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -914,9 +918,25 @@ public void run() {
                             forceWriteBatchEntriesStats.registerSuccessfulValue(toFlush.size());
                             forceWriteBatchBytesStats.registerSuccessfulValue(batchSize);
 
-                            forceWriteRequests.put(new ForceWriteRequest(logFile, logId, lastFlushPosition, toFlush,
-                                    (lastFlushPosition > maxJournalSize), false));
-                            toFlush = new LinkedList<QueueEntry>();
+                            boolean shouldRolloverJournal = (lastFlushPosition > maxJournalSize);
+                            if (syncData) {
+                                // Trigger data sync to disk in the "Force-Write" thread. Callback will be triggered after data is committed to disk
+                                forceWriteRequests.put(new ForceWriteRequest(logFile, logId, lastFlushPosition, toFlush, shouldRolloverJournal, false));
+                                toFlush = new LinkedList<QueueEntry>();
+                            } else {
+                                // Data is already written on the file (though it might still be in the OS page-cache)
+                                lastLogMark.setCurLogMark(logId, lastFlushPosition);
+                                for (int i = 0; i < toFlush.size(); i++) {
+                                    cbThreadPool.execute((QueueEntry) toFlush.get(i));
+                                }
+
+                                toFlush.clear();
+                                if (shouldRolloverJournal) {
+                                    forceWriteRequests.put(new ForceWriteRequest(logFile, logId, lastFlushPosition,
+                                            new LinkedList<>(), shouldRolloverJournal, false));
+                                }
+                            }
+
```

Comment:
Why not simply move the logic into JournalChnnel:forceWrite()?
I haven't really thought through any issues that may come up with the durability guarantees that were built into the system.

Reply:
I think this impl is good as it does not change other logics.
**We MUST add a strong advice that bookies with this flag are very dangerous as they does not guarantee a repeatable behaviour upon restart.**
They will eventually lose "fence bits" and will not guarantee the correctes of the LAC protocol, as writes are not guaranteed to have been really persisted to storage.

Reply:
> Why not simply move the logic into JournalChnnel:forceWrite()?

One reason is simplicity (the logic in the force-sync thread and request is already complicated, and I didn't want to add more in there). 

The other is to avoid one additional context switch before sending the ack. If we say that the write is "good" (for whatever definition of "good" is configured in the bookie), then we should acknowledge immediately.